src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,198 @@
+package duke;
+
+import java.util.ArrayList;
+import java.util.Scanner;
+import task.TaskList;
+import task.Task;
+import task.Command;
+import task.Parser;
+import task.Event;
+import task.Deadline;
+import task.Todo;
+import storage.Storage;
+import ui.Ui;
+
+public class Duke {
+    protected final TaskList taskList;
+    protected final ArrayList<String> commandList;
+    protected Parser parser;
+    protected Storage storage;
+    protected Ui ui;
+    protected String directoryPath;
+    protected String dataFilePath;
+
+    public Duke() {
+        this.dataFilePath = "./data/duke.txt";
+        this.directoryPath = "./data";
+        this.commandList = new ArrayList<>();
+        this.taskList = new TaskList();
+        this.parser = new Parser();
+        this.storage = new Storage(directoryPath, dataFilePath);
+        this.ui = new Ui();
+    }
+
+    // methods
+    public void addTask(Task task_given) {
+        this.taskList.add(task_given);
+    }
+
+    public String msg_of_add(Task task) {
```

Comment:
camelCase for method names can be used, would be better for both readability and the coding standard